crates/metadata-providers/src/objstore/glue.rs

```diff
@@ -116,7 +116,7 @@ impl Client {
 
 #[async_trait::async_trait]
 impl ProvisionedMetadataStore for Client {
-    #[instrument(level = "trace", skip_all, err)]
+    #[instrument(level = "trace", skip_all, err(level = "debug"))]
```

Comment:
We wouldn't see the "key" if an error has been logged since the span is emitted only at `trace` level. Perhaps make those spans `debug` as well?